src/main/scala/cognite/spark/v1/CdpConnector.scala

```diff
@@ -21,14 +22,23 @@ final case class Error[A](error: A)
 final case class Login(user: String, loggedIn: Boolean, project: String, projectId: Long)
 
 object CdpConnector {
+  private val logger = getLogger
   // It's important that the threads made here are daemon threads
   // so that we don't hang applications using our library during exit.
   // See for more info https://github.com/cognitedata/cdp-spark-datasource/pull/415/files#r396774391
   @transient lazy val cdpConnectorExecutionContext: ExecutionContext =
     ExecutionContext.fromExecutor(
       Executors.newFixedThreadPool(
         Math.max(Runtime.getRuntime.availableProcessors(), 4) * 2,
-        new ThreadFactoryBuilder().setDaemon(true).setNameFormat("CDF-Spark-Datasource-%d").build()
+        new ThreadFactoryBuilder()
+          .setDaemon(true)
+          .setUncaughtExceptionHandler(new UncaughtExceptionHandler {
+            override def uncaughtException(t: Thread, e: Throwable): Unit =
+              // Log the exception, and move on.
+              logger.warn(e)("Ignoring uncaught exception")
```

Comment:
Just to be sure I'm understanding this correctly, the `(e)` means we'll get stack traces and other exception details logged, right?

Reply:
Indeed!